Include mission statement on the home page (#423)
json-schema-org/json-schema-org.github.io: index.md

---

> **JSON Schema** enables the confident and reliable use of the JSON data format.

Reviewer: I think the original language here was a bit more concrete (i.e. corresponded directly to specific functionality). So maybe we leave both? E.g.

```suggestion
**JSON Schema** is a vocabulary that allows you to **annotate** and **validate** JSON documents. It enables the confident and reliable use of the JSON data format.
```

> **JSON Schema** is a vocabulary that allows you to **annotate** and **validate** JSON documents. It enables the confident and reliable use of the JSON data format.

Reviewer 2: I don't think "vocabulary" is the right word here. Maybe something more like this?

```suggestion
**JSON Schema** is a JSON document that allows you to define rules to **annotate** and **validate** JSON documents. It enables the confident and reliable use of the JSON data format.
```

Author: I would argue that "**A** JSON Schema" is a "JSON document", while "JSON Schema" the specification defines a vocabulary for use within "a JSON Schema JSON document"

Author: Also, this is not something that has changed in this PR.
This PR adds to that line, but mainly it's about the additional "mission" section.

Reviewer 2: I'm fine if you don't want to change this now. It's just something that bothers me every time I see it and this seemed like a good opportunity to fix it. What does "vocabulary" even mean in this context? I'm pretty sure it has nothing to do with JSON Schema Vocabularies, but I can't be sure. It's just way too generic a word to be meaningful.

As I seem to be saying a lot lately, I like to describe JSON Schema as a declarative domain specific language for validating and annotating JSON-compatible data. I suggest we used those words, but whatever you choose is fine even if you choose to do nothing for now and come back to it later.

Reviewer: It's being used in a non technical sense I suspect. And very likely predates any technical thing called vocabularies in JSON Schema. It doesn't bother me a ton but I agree it doesn't seem like the best way to describe us either so I'm definitely not opposed to changing it either.

> **JSON Schema** is a declarative domain specific grammar and syntax that allows you to **annotate** and **validate** JSON documents. JSON Schema enables the confident and reliable use of the JSON data format.

Reviewer 2: Isn't "grammar and syntax" just a more verbose way to say "language"? Also, domain specific language (DSL) is specific well defined technical term. Saying "language" in a different way shouldn't change the meaning, but it does increase the cognitive load of parsing this sentence. Of course it could just be that the sentence is easy to parse because I'm familiar with these technical terms. Not everyone is.

Reviewer 3: JSON Schema is way closer to a language than a grammar or syntax. Our grammar and syntax is entirely imported, and in no way domain-specific.

> Not everyone is.

Most of our intended audience should be familiar with the terms, but our front page should be written for the widest possible audience, not just our intended audience.

Author: Let's go with language then!

## Mission

> Our mission is to enable the definition of interoperable contracts for JSON data in the form of schemas, using constraints based validation rules, while enabling and supporting use cases beyond validation.

Reviewer: I find this more natural without the second comma -- 

```suggestion
Our mission is to enable the definition of interoperable contracts for JSON data in the form of schemas, using constraints based validation rules while enabling and supporting use cases beyond validation.
```

Content-wise it seems reasonable enough to me though so happy to thumbs up myself if the plan is to address Jason's point above in some other PR later.

Author: An editor friend of mine has confirmed that the comma is necessary. Otherwise the meaning sort of changes.

> Our mission is to enable the definition of interoperable contracts for JSON data in the form of schemas, using constraints based validation rules, while enabling and supporting use cases beyond validation.

Reviewer 4: I would like to add something to this that mentions annotation by name, because the spec should ensure interoperability around producing and consuming annotations.  The phrase "supporting use cases beyond validation" is vague and completely open-ended.  Annotations are _how_ additional use cases are supported, and that's what the JSON Schema project is concerned with.

Perhaps just this would do:

> Our mission is to enable the definition of interoperable contracts for JSON data in the form of schemas, using constraints-based validation rules, while enabling and supporting **_further use cases through annotation._**

I think that makes it clear that both validation and annotation are part of the interoperable contracts.

Author: Good point. Thinking on this though, I consider the use case where a schema is taken "as is" without an instance, and something is created, like a class or a database table (or 2). Are they still annotations?

I wonder if there's a new class of keywords for new vocabularies which are intended to be used without an instance in mind... maybe "declaration". I guess that could be seen as we're saying it's a "declarative" language. We have the duality being used both with and without an instance. Without an instance, it sort of becomes declarative, but still constraints based.

Reviewer 4: > Thinking on this though, I consider the use case where a schema is taken "as is" without an instance, and something is created, like a class or a database table (or 2). Are they still annotations?

Ooooh... excellent question!  Yeah, I've tended to think of that as a different way of making use of the existing keyword classes (not just annotations - assertions like `type` and applicators like `properties` are part of such cases as well).  But that way of declaring things with a schema is worth calling out explicitly.  I think of these as "generative" use cases (generating an instance from a schema- not necessarily generating a JSON instance but usually a code generator or whatever is still working with the JSON data model which is translated further into Java, SQL, etc.

Definitely worth further thought and clear language.
